### PR TITLE
Add docker stats support

### PR DESCRIPTION
**Pull request by patelrit**
Add docker stats support for docker 1.5+
It works in the same ways as events (async)
Also added a test case

@marcuslinke 
Please review and merge

### CODE REVIEW

**Comment by ericfjosne** [comment]
+1, would love to see this as mainstream !

File: src/test/java/com/github/dockerjava/client/AbstractDockerClientTest.java [hunk 1]
```diff
@@ -60,7 +60,7 @@ private DockerClientConfig config() {
 
     protected DockerClientConfig config(String password) {
         DockerClientConfig.DockerClientConfigBuilder builder = DockerClientConfig.createDefaultConfigBuilder()
-                .withServerAddress("https://index.docker.io/v1/");
+                .withServerAddress("https://index.docker.io/v1/").withMaxTotalConnections(5).withMaxPerRouteConnections(5);
```
Comment (marcuslinke) [inline src/test/java/com/github/dockerjava/client/AbstractDockerClientTest.java]: This should not be needed. Could you remove this please?

Comment (ericfjosne) [inline src/test/java/com/github/dockerjava/client/AbstractDockerClientTest.java]: Well spotted !

Done.

Thanks !
On 7 Apr 2015 at 14:23:49, marcuslinke (notifications@github.com) wrote:

In src/test/java/com/github/dockerjava/client/AbstractDockerClientTest.java:

> @@ -60,7 +60,7 @@ private DockerClientConfig config() {
>   
>      protected DockerClientConfig config(String password) {
>          DockerClientConfig.DockerClientConfigBuilder builder = DockerClientConfig.createDefaultConfigBuilder()
> -                .withServerAddress("https://index.docker.io/v1/");
> +                .withServerAddress("https://index.docker.io/v1/").withMaxTotalConnections(5).withMaxPerRouteConnections(5);
This should not be needed. Could you remove this please?

Comment (ericfjosne) [inline src/test/java/com/github/dockerjava/client/AbstractDockerClientTest.java]: Hi Markus,

My apologies, I’m still quite new to github.
I believe it should be correct now.

Best regards,

Eric

On 7 Apr 2015 at 14:40:24, Eric Fjøsne (eric.fjosne@gmail.com) wrote:

Well spotted !

Done.

Thanks !
On 7 Apr 2015 at 14:23:49, marcuslinke (notifications@github.com) wrote:

In src/test/java/com/github/dockerjava/client/AbstractDockerClientTest.java:

> @@ -60,7 +60,7 @@ private DockerClientConfig config() {
>    
>      protected DockerClientConfig config(String password) {
>          DockerClientConfig.DockerClientConfigBuilder builder = DockerClientConfig.createDefaultConfigBuilder()
> -                .withServerAddress("https://index.docker.io/v1/");
> +                .withServerAddress("https://index.docker.io/v1/").withMaxTotalConnections(5).withMaxPerRouteConnections(5);

This should not be needed. Could you remove this please?

**Comment by patelrit** [comment]
Is this ready to be merged or additional changes are required. Please let me know.

**Comment by marcuslinke** [comment]
@patelrit See https://github.com/docker-java/docker-java/issues/196 please.

As your implementation is derived from ```EventsCmdExec``` the same counts here. 

That's why the tests passes only when configuring a bigger connection pool.

So before i can merge i have to find out the cause of that problem.

**Comment by patelrit** [comment]
Thank you!

**Pull request merged**